pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -338,11 +338,24 @@ func getConfig(ctx *types.SystemContext) (*V2RegistriesConf, error) {
 	configPath := ConfigPath(ctx)
 
 	configMutex.Lock()
-	defer configMutex.Unlock()
 	// if the config has already been loaded, return the cached registries
 	if config, inCache := configCache[configPath]; inCache {
+		configMutex.Unlock()
 		return config, nil
 	}
+	configMutex.Unlock()
+
+	return TryUpdatingCache(ctx)
+}
+
+// TryUpdatingCache loads the registries from the provided
+// `SystemContext` without using the internal cache. On success, the found
+// registries will be added into the internal registry cache.
+func TryUpdatingCache(ctx *types.SystemContext) ([]Registry, error) {
```

Comment:
The code now returns `v2Config` value (which is a `V2RegistriesConf`), not `[]Registry`.

BTW it also seems to me that the `TryUpdatingCache` might want to return the full `V2RegistriesConf` value; the caller can always discard unwanted data but getting the desired data would otherwise be not quite as easy. OTOH the downside is that the callers, even callers that don’t care, might need to hard-code knowledge about the config file format. I don’t have a strong preference in this choice, just highlighting something to consider.

Reply:
Fixed the return value. I have no preference about the second point :)